main.go

```diff
@@ -251,3 +328,21 @@ func rawSetuid(uid int) error {
 
 	return nil
 }
+
+func (handler *BuildHandler) saveNewBuild(repoName string) BuildInfo {
+	info := BuildInfo{
+		repository: repoName,
+		startedAt:  time.Now(),
+		status:     "in progress",
+
+		pointer: handler.currentBuild,
+	}
+	handler.currentBuild.Value = info
+
+	if handler.currentBuild.Next() == handler.lastBuilds {
```

Comment:
Why not to use single pointer to the ring list? Every time you need to push new element, just do `pointer = pointer.Next()`, `pointer.Value = element`. For doing walk just do `pointer.Do()`.